pypeit/core/gui/object_find.py

```diff
@@ -0,0 +1,686 @@
+        # Replot, regardless of whether a fit is done (a point might have been added/removed)
+        self.replot()
+
+    def add_object(self):
+        """Add an object to specobjs
+        """
+        if self._trcmthd == 'manual' and len(self._mantrace['spat_a']) <= self._mantrace['polyorder']:
+            self.update_infobox(message="You need to select more trace points before manually adding\n" +
+                                        "a manual object trace. To do this, use the 'm' key", yesno=False)
+            return
+        # Add an object trace
+        spec_vec = self._mantrace['spec_trc']
+        if self._trcmthd == 'manual':
+            trace_model = self._mantrace['spat_trc'].copy()
+            # Now empty the manual tracing
+            self.empty_mantrace()
+        else:
+            trace_model = self._trcdict["trace_model"][self._trcmthd]["trace_model"].copy()
+            spat_0 = np.interp(self.mmy, spec_vec, trace_model)
+            shift = self.mmx - spat_0
+            trace_model += shift
+        xsize = self._trcdict["edges_r"] - self._trcdict["edges_l"]
+        nsamp = np.ceil(xsize.max())
+        # Extract the SpecObj parameters
+        par = self._trcdict['sobj_par']
+        # Create a SpecObj
+        thisobj = specobjs.SpecObj(par['frameshape'], par['slit_spat_pos'], par['slit_spec_pos'],
+                                   det=par['det'], setup=par['setup'], slitid=par['slitid'],
+                                   orderindx=par['orderindx'], objtype=par['objtype'])
+        thisobj.hand_extract_spat = self.mmx
+        thisobj.hand_extract_spec = self.mmy
+        thisobj.hand_extract_det = par['det']
+        thisobj.hand_extract_fwhm = None
+        thisobj.hand_extract_flag = True
+        f_ximg = RectBivariateSpline(spec_vec, np.arange(self.nspat), par["ximg"])
+        thisobj.spat_fracpos = f_ximg(thisobj.hand_extract_spec, thisobj.hand_extract_spat,
+                                      grid=False)  # interpolate from ximg
+        thisobj.smash_peakflux = np.interp(thisobj.spat_fracpos * nsamp, np.arange(nsamp),
+                                           self._trcdict['profile'])  # interpolate from fluxconv
+        # assign the trace
+        thisobj.trace_spat = trace_model
+        thisobj.trace_spec = spec_vec
+        thisobj.spat_pixpos = thisobj.trace_spat[self.nspec//2]
+        thisobj.set_idx()
+        if len(self.specobjs) != 0:
+            thisobj.fwhm = self.specobjs[0].fwhm
+        else:  # Otherwise just use the fwhm parameter input to the code (or the default value)
+            thisobj.fwhm = 2
+        # Finally, add new object
+        self.specobjs.add_sobj(thisobj)
+
+    def delete_object(self):
+        """Delete a specobj
+        """
+        self.specobjs.remove_sobj(self._obj_idx)
+        self._obj_idx = -1
+        self.replot()
+
+    def make_objprofile(self):
+        """Generate an object profile from the traces
```

Comment:
So if the WHT/ISIS detector does not include the slit boundaries because the entire detector is illuminated, then you need to use the standard as the crutch for slit tracing. The standard already gets passed to the object finding code to be used as a crutch, so I think this should be dealt with correctly, but it is worth checking.